modules/sdk-core/src/account-lib/mpc/tss/ecdsa/ecdsa.ts

```diff
@@ -229,6 +236,34 @@ export default class Ecdsa {
     return shares;
   }
 
+  /**
+   * Derives a child common keychain from common keychain
+   *
+   * @param commonKeychain - common keychain as a hex string
+   * @param path - bip32 path
+   * @return {string} derived common keychain as a hex string
+   */
+  deriveUnhardened(commonKeychain: string, path: string): string {
```

Comment:
Raising a flag here: We shouldn't be re-implementing BIP32. Just use `import { bip32 } from '@bitgo/utxo-lib';`

This applies to the code already added to `hdTree.ts`. That file should be moved down into an eddsa specific location. For ecdsa the functions are readily available in open source libraries we already depend on.

Reply:
This PR only implements a function to derive a child key from an existing hdTree. I don't have context behind why a BIP32 class was implemented for ECDSA instead of using utxo-lib https://github.com/BitGo/BitGoJS/pull/3029/files
Maybe @johnoliverdriscoll or @sachushaji can provide context here

Reply:
For TSS we need to derive the public and private keys separately. The user can derive the common key themselves in standard BIP32, but each level of derivation must be added to a *share* of the private key, not the full private key (which never exists in a fully constructed form).

Reply:
Just to explicitly state it, existing open source libraries are not compatible with BIP32 derivation for TSS shares.

Reply:
The unhardened derivation is identical. Seems like the cleaner solution would be to fork the `bip32` library and add the slight difference (which appears to be only that the `sk` portion isn't directly the sk for the combined key) as a mode there, rather than taking on maintenance of a separate bip32 implementation.

Reply:
I don't think forking a much larger library that includes features like serialization and fingerprinting that we don't use here would be a cleaner solution than implementing the simple BIP-32 derivation algorithm. Unless you expect the decade old BIP to be revised anytime soon I don't understand this being an issue of maintenance coste.

Reply:
It's a 336 line (by sloccount on the .js) library, and we already have forks of several others from the bitcoinjs team (`ecpair`, `bip174`, and `bitcoinjs-lib`).

Reply:
Either way I think this is no longer a relevant argument for this PR. There is no way to use an existing BIP-32 implementation as is because of the secret key derivation for shares. If you want the existing implementation that *is* compatible with TSS scrapped in favor of making an entirely new dependency for this single task, you should make a ticket for it.